### PR TITLE
Add node8-typescript sample

### PR DESCRIPTION
**Pull request by dakale**
Adds a sample YAML build consuming the node8-typescript container

Sample project from: https://github.com/Microsoft/TypeScript-Node-Starter

azure-pipelines.yml contains build sample, rest is sample project to build

cc @bryanmacfarlane @vtbassmatt

### CODE REVIEW

File: containers/samples/node8-typescript/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+resources:
```
Comment (ericsciple) [inline containers/samples/node8-typescript/azure-pipelines.yml]:
```yaml
variables:
  sampleRoot: containers/samples/node8-typescript/TypeScript-Node-Starter
jobs:
- jobs: foo
  steps:
  - script: asdf
    workingDirectory: $(sampleRoot)
```

File: containers/samples/node8-typescript/azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  - task: Npm@1
+    displayName: 'npm run build'
+    inputs:
+      command: custom
+      customCommand: 'run build'
+      workingDir: containers/samples/node8-typescript/TypeScript-Node-Starter
+      verbose: false
+
+  - script: |
+      echo "Copying built dist into container (pseudo-multistage)"
```
Comment (ericsciple) [inline containers/samples/node8-typescript/azure-pipelines.yml]: avoid abbreviation?

File: containers/samples/node8-typescript/azure-pipelines.yml (hunk 1 above)
Comment (dakale) [reply containers/samples/node8-typescript/azure-pipelines.yml]: 👍

File: containers/samples/node8-typescript/azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    displayName: Check docker images
+
+- job: 'In_node8_typescript_build_container_multistage'
+  container: node8-typescript
+  pool:
+    vmImage: 'Ubuntu-16.04'
+
+  steps:
+  - script: |
+      echo "Building in one command (real multistage, from container)"
```
Comment (ericsciple) [inline containers/samples/node8-typescript/azure-pipelines.yml]: can we avoid the "stage" term? or is this specific to the domain?

**Comment by dakale** [comment]
This work moved to https://github.com/Microsoft/azure-pipelines-canary/tree/master/src/node-typescript

**Pull request closed without merging**